File: app/src/main/java/com/m6world/tari_labs/api/_interactors/EventsInteractor.java
package com.m6world.tari_labs.api._interactors;

import com.m6world.tari_labs.api._contracts.IEventsInteractor;
import com.m6world.tari_labs.api.models.EventCheckinsResponse;
import io.reactivex.Observable;
import org.jetbrains.annotations.NotNull;

import java.util.HashMap;
import java.util.Map;

public class EventsInteractor implements IEventsInteractor {
    GetNetworkInteractor getNetworkInteractor = new GetNetworkInteractor();

    private final String END_POINT = "events/checkins";

    @NotNull
    @Override
    public Observable<EventCheckinsResponse> getEventCheckins(@NotNull String token) {
        Map<String, String> headerMap = new HashMap<>();
        headerMap.put("Authorization", "Bearer " + token);
        return getNetworkInteractor.createObservable(END_POINT, "GetEventCheckins", headerMap, EventCheckinsResponse.class);
    }
}

File: app/src/main/java/com/m6world/tari_labs/api/interactors/EventsInteractor.kt
package com.m6world.tari_labs.api.interactors

import com.m6world.tari_labs.api.contracts.IEventsInteractor
import com.m6world.tari_labs.api.models.EventCheckinsResponse
import io.reactivex.Observable

import java.util.HashMap

class EventsInteractor : IEventsInteractor {
    internal var getNetworkInteractor = GetNetworkInteractor()

    private val END_POINT = "events/checkins"

    override fun getEventCheckins(token: String?): Observable<EventCheckinsResponse> {
        val headerMap = HashMap<String, String>()
        headerMap["Authorization"] = "Bearer $token"
        return getNetworkInteractor.createObservable(
            END_POINT,
            "GetEventCheckins",
            headerMap,
            EventCheckinsResponse::class.java
        )
    }
}